e2e/tests/contracts.rs

```diff
@@ -2105,3 +2105,177 @@ async fn max_fee_estimation_respects_tolerance() -> Result<()> {
+         contract_size > contract_max_size,
+         "this test should ideally be run with a contract bigger than the max contract size ({contract_max_size}B) so that we know deployment couldn't have happened without blobs"
+     );
+
+    let contract = Contract::load_from(contract_binary, LoadConfiguration::default())?;
+
+    let contract_id = contract
+        .convert_to_loader(100_000)?
+        .deploy(&wallets[0], TxPolicies::default())
+        .await?;
+
+    let contract_instance = MyContract::new(contract_id, wallets[0].clone());
+
+    let response = contract_instance.methods().something().call().await?.value;
+
+    assert_eq!(response, 1001);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn regular_contract_can_be_deployed() -> Result<()> {
+    // given
+    setup_program_test!(
+        Wallets("wallet"),
+        Abigen(Contract(
+            name = "MyContract",
+            project = "e2e/sway/contracts/contract_test"
+        )),
+    );
+
+    let contract_binary = "sway/contracts/contract_test/out/release/contract_test.bin";
+
+    // when
+    let contract_id = Contract::load_from(contract_binary, LoadConfiguration::default())?
+        .deploy(&wallet, TxPolicies::default())
+        .await?;
+
+    // then
+    let contract_instance = MyContract::new(contract_id, wallet);
+
+    let response = contract_instance
+        .methods()
+        .get_counter()
+        .call()
+        .await?
+        .value;
+
+    assert_eq!(response, 0);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn unuploaded_loader_can_be_deployed_directly() -> Result<()> {
+    setup_program_test!(
+        Wallets("wallet"),
+        Abigen(Contract(
+            name = "MyContract",
+            project = "e2e/sway/contracts/huge_contract"
+        )),
+    );
+
+    let contract_binary = "sway/contracts/huge_contract/out/release/huge_contract.bin";
+
+    let contract_id = Contract::load_from(contract_binary, LoadConfiguration::default())?
+        .convert_to_loader(1024)?
+        .deploy(&wallet, TxPolicies::default())
+        .await?;
+
+    let contract_instance = MyContract::new(contract_id, wallet);
+
+    let response = contract_instance.methods().something().call().await?.value;
+
+    assert_eq!(response, 1001);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn unuploaded_loader_can_upload_blobs_separately_then_deploy() -> Result<()> {
+    setup_program_test!(
+        Wallets("wallet"),
+        Abigen(Contract(
+            name = "MyContract",
+            project = "e2e/sway/contracts/huge_contract"
+        )),
+    );
+
+    let contract_binary = "sway/contracts/huge_contract/out/release/huge_contract.bin";
+
+    let contract = Contract::load_from(contract_binary, LoadConfiguration::default())?
+        .convert_to_loader(1024)?
+        .upload_blobs(&wallet, TxPolicies::default())
+        .await?;
+
+    let blob_ids = contract.blob_ids();
+
+    // if this were an example for the user we'd just call `deploy` on the contract above
```

Comment:
```suggestion
    // if this was an example for the user we'd just call `deploy` on the contract above
```

Reply:
wanted to say "had this been an example", so were would be closer here.

TIL it's called [subjunctive mood](https://www.grammar-monster.com/glossary/subjunctive_mood.htm)